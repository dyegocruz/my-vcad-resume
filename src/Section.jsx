import React from 'react';
import PropTypes from 'prop-types';

class Section extends React.PureComponent {
  render() {
    const { variant, id, children } = this.props;
    return (
      <section className={variant} id={id}>
        {children}
      </section>
    );
  }
}

Section.propTypes = {
  variant: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Section;
