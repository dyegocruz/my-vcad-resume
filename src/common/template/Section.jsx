import React from 'react';

const Section = ({ variant, id, children }) => (
  <section className={variant} id={id}>
    {children}
  </section>
);

export default Section;
