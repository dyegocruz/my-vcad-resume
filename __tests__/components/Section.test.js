import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Section from '../../src/common/template/Section';

configure({ adapter: new Adapter() });

// Snapshot for Section React Component
describe('Section Component Snapshot', () => {
  it('+++capturing Snapshot of Home', () => {
    const renderedValue = renderer.create(<Section />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('Shallow Render Section React Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Section />);
  });

  it('Should be render Section component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('Should contains section html element', () => {
    expect(wrapper.find('section')).toHaveLength(1);
  });
});
