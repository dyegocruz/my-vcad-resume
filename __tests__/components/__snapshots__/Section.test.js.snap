// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component Snapshot +++capturing Snapshot of Home 1`] = `<section />`;
